manticore/core/cpu/arm.py

```diff
@@ -838,6 +906,19 @@ def CLZ(cpu, dest, src):
     def NOP(cpu):
         pass
 
+    @instruction
+    def REV(cpu, dest, op):
+        opval = op.read()
+        _bytes = list()
+        for i in range(4):
+            _bytes.append(Operators.EXTRACT(opval, i*8, 8))
+        dest.write(Operators.CONCAT(32, *_bytes))
+
+    @instruction
+    def SXTH(cpu, dest, op):
+        _op = op.read()
+        dest.write(Operators.SEXTEND(Operators.EXTRACT(_op, 0, 16), 16, 32))
```

Comment:
The documentation I was referencing didn't mention the rotate option, as it turns out. I added a test, and this works just fine. (I'm completely surprised and delighted.)